tests/test_unit/brainglobe_workflows/test_cellfinder.py

```diff
@@ -0,0 +1,343 @@
+import json
+import logging
+import re
+from pathlib import Path
+
+import pooch
+import pytest
+
+from brainglobe_workflows.cellfinder import (
+    CellfinderConfig,
+    add_signal_and_background_files,
+    read_cellfinder_config,
+    run_workflow_from_cellfinder_run,
+    setup_workflow,
+)
+from brainglobe_workflows.cellfinder import setup as setup_full
+from brainglobe_workflows.utils import setup_logger
+
+
+@pytest.fixture()
+def default_input_config_cellfinder() -> Path:
+    """Return path to default input config for cellfinder workflow
+
+    Returns
+    -------
+    Path
+        Path to default input config
+
+    """
+    from brainglobe_workflows.utils import DEFAULT_JSON_CONFIG_PATH_CELLFINDER
+
+    return DEFAULT_JSON_CONFIG_PATH_CELLFINDER
+
+
+@pytest.mark.parametrize(
+    "input_config",
+    [
+        "input_data_GIN.json",
+        "input_data_locally.json",
+        "input_data_missing_background.json",
+        "input_data_missing_signal.json",
```

Comment:
Could these two be replaced by a separate test function and mocks returning missing signal/background instead of needing a whole repeated json file each?

Reply:
if that is alright I will leave this as is for now (to not further bloat this chubby PR), but I have an issue for improving tests with mocking and caching #34